### PR TITLE
Manage support requests admin section

### PR DESCRIPTION
**Pull request by xavierdecoster**

### CODE REVIEW

File: src/NuGetGallery/App_Start/DefaultDependenciesModule.cs [hunk 1]
```diff
@@ -124,6 +126,21 @@ protected override void Load(ContainerBuilder builder)
...
+                .AsSelf()
+                .As<ISupportRequestDbContext>()
+                .InstancePerLifetimeScope();
+
+            builder.RegisterType<SupportRequestService>()
+                .AsSelf()
+                .As<ISupportRequestService>()
+                .InstancePerLifetimeScope();
+
+            builder.RegisterType<PagerDutyService>()
```
Comment (xavierdecoster) [inline src/NuGetGallery/App_Start/DefaultDependenciesModule.cs]: In a next phase, I'd like to move this dependency external to the gallery.

File: src/NuGetGallery/Areas/Admin/Views/SupportRequest/Index.cshtml [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+
+@section BottomScripts {
+    @if (Model != null)
+    {
+        <script src="~/Scripts/jquery-ui-1.10.3.min.js"></script>
+        <script src="~/Scripts/moment.min.js"></script>
+        <script src="~/Scripts/knockout-2.2.1.js"></script>
+        <script src="~/Scripts/knockout.mapping-latest.js"></script>
+
+        <script type="text/javascript">
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Views/SupportRequest/Index.cshtml]: Any chance the models can go into separate .js files?

File: src/NuGetGallery/NuGetGallery.csproj [hunk 3]
```diff
@@ -1748,7 +1793,7 @@
           <AutoAssignPort>True</AutoAssignPort>
           <DevelopmentServerPort>80</DevelopmentServerPort>
           <DevelopmentServerVPath>/</DevelopmentServerVPath>
-          <IISUrl>https://nuget.localtest.me</IISUrl>
+          <IISUrl>http://nuget.localtest.me</IISUrl>
```
Comment (maartenba) [inline src/NuGetGallery/NuGetGallery.csproj]: Can this remain https:// ?

**Comment by maartenba** [comment]
:shipit: once nits fixed

File: src/NuGetGallery/Areas/Admin/Views/SupportRequest/Index.cshtml (hunk 2 above)
Comment (xavierdecoster) [inline src/NuGetGallery/Areas/Admin/Views/SupportRequest/Index.cshtml]: Not with the embedded Razor syntax that binds the JS viewmodel to the MVC viewmodel.

File: src/NuGetGallery/NuGetGallery.csproj (hunk 3 above)
Comment (xavierdecoster) [inline src/NuGetGallery/NuGetGallery.csproj]: Yes, no idea why that was changed :) Addressed with https://github.com/NuGet/NuGetGallery/commit/a770f0341d6d6642fe529a9e307a589453bfcc93

File: src/NuGetGallery/Areas/Admin/Views/SupportRequest/Index.cshtml (hunk 2 above)
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Views/SupportRequest/Index.cshtml]: You can pass those in as parameters to the constructor of your model?
```var viewModel = new ViewModel('@Url......', 'etc.');```

**Comment by xavierdecoster** [comment]
Fixed with https://github.com/NuGet/NuGetGallery/commit/3db741b9f5c865acef2520aae60c2c7e3df2122f

File: src/NuGetGallery/Areas/Admin/Views/SupportRequest/Index.cshtml (hunk 2 above)
Comment (xavierdecoster) [inline src/NuGetGallery/Areas/Admin/Views/SupportRequest/Index.cshtml]: Done

**Pull request closed without merging**